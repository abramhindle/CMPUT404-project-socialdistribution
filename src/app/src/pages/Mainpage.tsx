import NavBar from "../components/NavBar";
import UserPost from "../components/UserPost";
import styled from "styled-components";
import Button, { ButtonProps } from "@mui/material/Button";
import { styled as Styled } from "@mui/material/styles";
import Author from "../api/models/Author";
import Post from "../api/models/Post";
import api from "../api/api";
import { useState, useEffect } from 'react';

const items2 = [
  {
    Text: "Hello",
    handleClick: () => {
      console.log(1);
    },
  },
];

// This is for all the stuff in the Main Page
const MainPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  height: ${window.innerHeight}px;
  width: ${window.innerWidth}px;
`;

// This is for the NavBar
const NavBarContainer = styled.div`
  margin-bottom: 5%;
`;

// This is for the posts and New Post Button and Github Activity
const MainPageContentContainer = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`;

// This is the NewPost Button
const NewPostButton = Styled(Button)<ButtonProps>(({ theme }) => ({
  color: theme.palette.getContrastText("#e6c9a8"),
  backgroundColor: "white",
  border: "2px solid black",
  height: "10%",
  width: "90%",
  padding: "1%",
  "&:hover": {
    backgroundColor: "#F9F7F5",
  },
}));

// This holds the New Post button and Github Stream
const NewPostGithubActivityContainer = styled.div`
  width: 10%;
  display: flex;
  justify-content: column;
  margin-right: 50px;
`;

interface Props {
  currentUser?: Author;
}

export default function Mainpage({ currentUser }: Props) {
  // For now, mainpage just shows your own posts
  const [posts,setPosts]= useState<Post[]|undefined>(undefined)
  
  useEffect(() => {
    api.authors
    .withId(""+currentUser?.id)
    .posts
    .list()
    .then((data)=>setPosts(data))
    .catch((error) => {console.log(error)})
    }, [currentUser?.id,posts])

  return (
    <MainPageContainer>
      <NavBarContainer>
        <NavBar items={items2} />
      </NavBarContainer>

      <MainPageContentContainer>
        {posts?.map((post) => (
          <UserPost
              Name={"Oogway"}
              ContentText={"Po is the dragon warrior"}
              Likes={10}
              Comments={6}
              key={post.id}
            />
          ))}

        <NewPostGithubActivityContainer>
          <NewPostButton>New Post</NewPostButton>
        </NewPostGithubActivityContainer>
      </MainPageContentContainer>
    </MainPageContainer>
  );
}
