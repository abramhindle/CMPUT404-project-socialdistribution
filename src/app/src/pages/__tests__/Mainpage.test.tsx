import React from "react";
import { unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";

import Mainpage from "../Mainpage";

let container: any = null;
//issues with container type and its methods

beforeEach(() => {
  // setup a DOM element as a render target
  container = document?.createElement("div");
  document?.body?.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container?.remove();
  container = null;
});

describe("Mainpage suite", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<Mainpage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
