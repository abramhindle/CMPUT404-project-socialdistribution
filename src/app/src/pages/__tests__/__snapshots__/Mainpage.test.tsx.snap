// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainpage suite Renders correctly 1`] = `
<div
  className="sc-iqseJM jtHRij"
>
  <div
    className="sc-crHmcD bQNHbF"
  >
    <div
      className="sc-bdvvtL dVmOGZ"
    >
      <div
        className="sc-gsDKAQ dokASy"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-79h83g-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Hello
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-egiyK QHpdd"
  >
    <div
      className="sc-dkPtRN CNuOq"
    >
      <div
        className="sc-eCImPb gCdvCh"
      >
        <img
          alt="Profile"
          height="100"
          width="100"
        />
      </div>
      <div
        className="sc-hKwDye YhLWm"
      >
        <div
          className="sc-jRQBWg cztAja"
        >
          <div
            className="sc-gKclnd iLxqaW"
          >
            Oogway
          </div>
          <div
            className="sc-iCfMLu cefipO"
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1wgo97p-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Edit
            </button>
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1htnu9u-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="sc-furwcr ckzOwp"
        >
          Po is the dragon warrior
        </div>
        <div
          className="sc-pVTFL dByDtN"
        >
          <div
            className="sc-jrQzAO kWwnDx"
          >
            10
             Likes
          </div>
          <div
            className="sc-kDTinF cITOvt"
          >
            6
             Comments
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bqiRlB OLywK"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1phbry0-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        New Post
      </button>
    </div>
  </div>
</div>
`;
